tapiriik/services/Pulsstory/pulsstory.py

```diff
@@ -0,0 +1,385 @@
+
+            # By default, use the provided waypoint types
+            wp_type_iter = (wp.Type for wp in flat_wps)
+            # Unless those types don't include pause/resume, in which case use our auto-pause calculation
+            if auto_pause and not any(wp.Type == WaypointType.Pause for wp in flat_wps):
+                # ...but not if we don't know the intended moving time
+                if activity.Stats.MovingTime.Value:
+                    wp_type_iter = AutoPauseCalculator.calculate(flat_wps, activity.Stats.MovingTime.asUnits(ActivityStatisticUnit.Seconds).Value)
+
+            inPause = False
+            for waypoint, waypoint_type in zip(flat_wps, wp_type_iter):
+                timestamp = (waypoint.Timestamp - anchor_ts).total_seconds()
+                record["Waypoints"]["TimeList"].append(timestamp)
+                
+                '''
+                if waypoint_type in self._wayptTypeMappings.values():
+                    wpType = [key for key, value in self._wayptTypeMappings.items() if value == waypoint_type][0]
+                else:
+                    wpType = "gps"  # meh
+                '''
+
+                if not inPause and waypoint_type == WaypointType.Pause:
+                    inPause = True
+                elif inPause and waypoint_type == WaypointType.Pause:
+                    continue
+                elif inPause and waypoint_type != WaypointType.Pause:
+                    inPause = False
+
+                if waypoint.HR is not None:
+                    record["Waypoints"]["HeartRateValue"].append(round(waypoint.HR))
+
+                if waypoint.Power is not None:                    
+                    record["Waypoints"]["PowerValue"].append(waypoint.Power)
+                    
+                if waypoint.Speed is not None:                    
+                    record["Waypoints"]["SpeedValue"].append(waypoint.Speed)
+                    
+                if waypoint.Cadence is not None:                    
+                    record["Waypoints"]["CadanceValue"].append(waypoint.Cadence)
+                    
+                if waypoint.Location is not None and waypoint.Location.Longitude is not None:                    
+                    record["Waypoints"]["LongitudePathValue"].append(waypoint.Location.Longitude)
+                    
+                if waypoint.Location is not None and waypoint.Location.Latitude is not None:
+                    record["Waypoints"]["LatitudePathValue"].append(waypoint.Location.Latitude)
+                
+                if waypoint.Location is not None and waypoint.Location.Altitude is not None:                 
+                    record["Waypoints"]["AltitudePathValue"].append(waypoint.Location.Altitude)
+
+        return record
+
+    def DeleteCachedData(self, serviceRecord):
+        cachedb.rk_activity_cache.remove({"Owner": serviceRecord.ExternalID})
+
+    def DeleteActivity(self, serviceRecord, uri):
```

Comment:
Leave implemented so an exception gets thrown if something were to attempt to call this, rather than failing silently.